docs/1.1-DRAFT/workflows.md

```diff
@@ -169,26 +175,70 @@ A workflow diagram may still be provided even if there is no programmatic `Softw
+* [targetProduct] to indicate a [SoftwareApplication] for the operating system or product which this code applies to, e.g. `https://www.microsoft.com/en-gb/windows/get-windows-10` or `https://releases.ubuntu.com/20.04/`
+
+The [ComputationalWorkflow] data entity MAY also describe, where relevant:
+* [alternateName] to indicate an alternative [name] that help identify the workflow, e.g. `nf-core/chipseq`
+* [conditionsOfAccess] to indicate restrictions on the availability of this workflow, e.g. `"To protect endangered wildlife, this biodiversity workflow is only available to approved research projects"`
+* [dateModified] to indicate a date later than [dateCreated] when this workflow was modified
+* [datePublished] to indicate a date different from [dateCreated] for when this workflow was made public
+* [encodingFormat] to indicate the [IANA media type](data-entities.md#adding-detailed-descriptions-of-encodings), e.g. `"application/xml"` or `"application/json"` (see also [programmingLanguage] above)
+* [image] to indicate an [ImageObject] that represents the workflow, e.g. a workflow diagram
+* [identifier] to provide a unique identifier for this workflow, e.g. `https://doi.org/10.5281/zenodo.3966161` or `urn:uuid:0a98135b-3e48-4f76-aed8-4a7f6421bd74`
+
+### Describing inputs and outputs
+
+The input and output _parameters_ for a workflow or script can be given with `input` and `output` to [FormalParameter]
+contextual entities. Note that this entity usually represent a _potential_ input/output value in a reusable
+workflow, much like [function parameter definitions](https://en.wikipedia.org/wiki/Parameter_(computer_programming))
+in general programming.
+
+When complying with the
+[BioSchemas FormalParameter profile](https://bioschemas.org/profiles/FormalParameter/0.1-DRAFT-2020_06_23/)
+the _contextual entities_ for [FormalParameter], referenced by `input` or `output`, MUST describe:
 
 * [name] given the programmatic name for the parameter binding
 * [additionalType] identifying the most specific subtype of [EDAM Data](http://edamontology.org/data_0006) (fallbacks [Data](http://edamontology.org/data_0006) or [Text data](http://edamontology.org/data_2526))
 * [encodingFormat] identifying the most specific subtype of [EDAM Format](http://edamontology.org/format_1915) (fallbacks [Binary format](http://edamontology.org/format_2333) or [Textual format](http://edamontology.org/format_2330))
+
+
+In addition the `FormalParameter` contextual entities SHOULD also include:
+* [description] describing the parameter's expected value range and purpose
+
+Contextual entities of type `FormalParameter` MAY also include any of:
 * [valueRequired] `true` if this (input) parameter must be specified to run the workflow, or `false` (default) if parameter is optional.
 * [defaultValue] present if this (input) parameter has a default value. In RO-Crate this SHOULD be in the form of a `"string"` or a `{"@id": "data/entity.txt"}`
```

Comment:
Why not add minimum and maximum accepted values? (Maybe in a future iteration?)

Reply:
They would only realistically apply to numeric parameters. It would need more thought - and yes a change to bioschemas

Reply:
In our work, having the following properties for parameters has been incredibly helpful:
- minimum value (min numeric value or date)
- maximum value (max numeric value or date)
- accepted values (i.e., this parameter is a string that accepts "Maize" and "Sorghum" as values)
- interval (i.e., create an ensemble increasing this parameter by 0.1 until the max value)